### DISCUSSION TITLE
iOS devices issue with retrieving wallet address

### DISCUSSION THREAD

**Discussion by alaabadran** [Web3Auth/Web3Auth#856] [Help]
We have integrated Web3Auth with our website.
Previously, when trying to login with iOS devices using Chrome browser.. It was working fine.
Lately, login is successful but wallet address is not retrieved. This is causing many functionalities to break up. And currently, I can make it work on Safari iOS browser by disabling `Prevent Cross-Site Tracking` in settings.

Debugging the issue (session storage data) on Safari Browser (iOS):
`{"userModule":{"userInfo":{"email":"","name":"","profileImage":"","verifier":"","aggregateVerifier":"","verifierId":"","typeOfLogin":"google"},"clientTimeOffset":0,"keyInfo":{"typeOfUser":"v1","privateKey":"","pubKey":{"pub_key_X":"","pub_key_Y":""},"publicAddress":"","metadataNonce":""},"walletKeyInfo":{"typeOfUser":"v1","privateKey":"","pubKey":{"pub_key_X":"","pub_key_Y":""},"publicAddress":"","metadataNonce":""},"currentDappClientId":"REMOVED","authToken":"","challenge":"","uniqueDapps":[],"alwaysSkip":false,"persistedUserInfo":{}}}`
All are empty

The data is blocked by the brower.
Is there any solution for this?

**Comment by himanshuchawla009** [comment]
If session storage is blocked then we fallback to an encrypted server channel. It should always return the address. You can DM me your app link and way to reproduce the issue on telegram: himanshuchawla009

**Reply by alaabadran** [reply]
I have sent you the app link to your Telegram.. Looking forward to your response.
Thanks

**Reply by alaabadran** [reply]
@himanshuchawla009 Just following up on this issue.. Have you go the chance to look into this?

**Comment by alaabadran** [comment]
I am not getting a support from @himanshuchawla009 Just getting generic answers.
Web3auth wallet is blocked by many browsers, and the fallback they talk about is not working.
Web3auth wallet and private key are blocked by Chrome iOS, Firefox, Brave, Safari browsers. Some works if you disable privacy protection.. Which is risky.
We need a fix to this issue please

**Reply by himanshuchawla009** [reply]
@yashovardhan @shahbaz17  can you help him here.

**Comment by yashovardhan** [comment]
Hey @alaabadran 

Can you please check your iOS version? Web3Auth cannot work with iOS v13 and lower. It only supports iOS 14 and above. This is because the browsers don't support BigInt.

Please check the compatible browsers here: https://browsersl.ist/#q=supports+bigint%2C+not+dead

**Reply by himanshuchawla009** [reply]
Please schedule a call with @alaabadran to help him out.

**Reply by yashovardhan** [reply]
@alaabadran please DM me on telegram `@yashweb3`

I'll send you the link for scheduling a call.

**Comment by yashovardhan** [comment]
This issue was solved by migrating to the latest SDK.

**Reply by alaabadran** [reply]
Thank you @yashovardhan
